prime-router/src/main/kotlin/serializers/CsvSerializer.kt

```diff
@@ -75,81 +75,70 @@ class CsvSerializer(val metadata: Metadata) : Logging {
                 readAllWithHeaderAsSequence().forEach { row: Map<String, String> ->
                     rows.add(row)
                     if (rows.size > REPORT_MAX_ITEMS) {
-                        errors.add(
-                            ResultDetail.report(
-                                InvalidReportMessage.new(
-                                    "Your file's row size of ${rows.size} exceeds the maximum of $REPORT_MAX_ITEMS " +
-                                        "rows per file. Reduce the amount of rows in this file."
-                                )
+                        throw ActionError(
+                            ActionLog.report(
```

Comment:
Why not simplify these calls and just pass in the string?

Reply:
Mostly because the scope or type of the log could be different. We could pass those through and have the `ActionError` construct the log but that doesn't seem like an improvement.